code/game/gamemodes/clock_cult/clock_objects.dm

```diff
@@ -0,0 +1,1579 @@
+///////////////////////
+
+/obj/effect/clockwork
+	name = "meme machine"
+	desc = "Still don't know what it is."
+	var/clockwork_desc = "A fabled artifact from beyond the stars. Contains concentrated meme essence." //Shown to clockwork cultists instead of the normal description
+	icon = 'icons/effects/clockwork_effects.dmi'
+	icon_state = "ratvars_flame"
+	anchored = 1
+	density = 0
+	opacity = 0
+
+/obj/effect/clockwork/New()
+	..()
+	all_clockwork_objects += src
+
+/obj/effect/clockwork/Destroy()
+	all_clockwork_objects -= src
+	return ..()
+
+/obj/effect/clockwork/examine(mob/user)
+	if(is_servant_of_ratvar(user) && clockwork_desc)
+		desc = clockwork_desc
+	..()
+	desc = initial(desc)
+
+/obj/effect/clockwork/judicial_marker //Judicial marker: Created by the judicial visor. After four seconds, stuns any non-servants nearby and damages Nar-Sian cultists.
+	name = "judicial marker"
+	desc = "You get the feeling that you shouldn't be standing here."
+	clockwork_desc = "A sigil that will soon erupt and smite any unenlightened nearby."
+	icon = 'icons/effects/96x96.dmi'
+	pixel_x = -30
+	pixel_y = -30
+	layer = 2.01
+
+/obj/effect/clockwork/judicial_marker/New()
+	..()
+	flick("judicial_marker", src)
+	spawn(25) //Utilizes spawns due to how it works with Ratvar's flame
+		flick("judicial_explosion", src)
+		spawn(15)
+			for(var/mob/living/L in range(1, src))
+				if(is_servant_of_ratvar(L))
+					continue
+				if(!iscultist(L))
+					L << "<span class='userdanger'>[!issilicon(L) ? "An unseen force slams you into the ground!" : "ERROR: Motor servos disabled by external source!"]</span>"
+					L.Weaken(8)
+				else
+					L << "<span class='heavy_brass'>\"Keep an eye out, filth.\"</span>\n<span class='userdanger'>[!issilicon(L) ? "An unseen force piledrives you into the ground!" : "ERROR: Motor servos damaged by external source!"]</span>"
+					L.Weaken(10)
+					L.adjustBruteLoss(10)
+			qdel(src)
+			return 1
+
+/obj/effect/clockwork/spatial_gateway //Spatial gateway: A one-way rift to another location.
+	name = "spatial gateway"
+	desc = "A gently thrumming tear in reality."
+	clockwork_desc = "A gateway in reality. It can either send or receive, but not both."
+	icon_state = "spatial_gateway"
+	density = 1
+	var/sender = TRUE //If this gateway is made for sending, not receiving
+	var/lifetime = 25 //How many deciseconds this portal will last
+	var/uses = 1 //How many objects or mobs can go through the portal
+	var/obj/effect/clockwork/spatial_gateway/linked_gateway //The gateway linked to this one
+
+/obj/effect/clockwork/spatial_gateway/New()
+	..()
+	spawn(1)
+		if(!linked_gateway)
+			qdel(src)
+			return 0
+		else
+			if(linked_gateway.sender == sender)
+				linked_gateway.sender = !sender
+		spawn(lifetime)
+			if(src)
+				qdel(src)
+
+/obj/effect/clockwork/spatial_gateway/Destroy()
+	visible_message("<span class='warning'>[src] flickers and closes!</span>")
+	animate(src, transform = matrix() - matrix(), time = 10)
+	spawn(10)
+		..()
```

Comment:
return ..()
also i'm not totally sure spawning() in this is safe, maybe spawn an object to do the effect for you?